pkg/service/service.go

```diff
@@ -1549,10 +1549,14 @@ func (s *Service) restoreServicesLocked(svcBackendsById map[lb.BackendID]struct{
 		}
 
 		for j, backend := range svc.Backends {
+			// DumpServiceMaps() can return services with some empty (nil) backends.
+			if backend == nil {
+				continue
+			}
+
 			hash := backend.L3n4Addr.Hash()
 			s.backendRefCount.Add(hash)
-			newSVC.backendByHash[hash] = &svc.Backends[j]
-			svcBackendsById[backend.ID] = struct{}{}
```

Comment:
`svcBackendsById[backend.ID] = struct{}{}` was not removed in the upstream PR ?

Reply:
👍 Great catch, this shouldn't have been removed. Added back.